src/Core/Services/SyncService.cs

```diff
@@ -382,5 +383,39 @@ private async Task SyncSendsAsync(string userId, List<SendResponse> response)
                 new Dictionary<string, SendData>();
             await _sendService.ReplaceAsync(sends);
         }
+
+        private async Task SyncPasswordlessLoginRequestsAsync(string userId)
+        {
+            // if the user has not enabled passwordless logins ignore requests
+            if (!await _stateService.GetApprovePasswordlessLoginsAsync(userId))
+            {
+                return;
+            }
+
+            var loginRequests = await _apiService.GetAuthRequestAsync();
+            if (loginRequests == null || !loginRequests.Any())
+            {
+                return;
+            }
+
+            var validLoginRequest = loginRequests.Where(l => l.RequestApproved == null
```

Comment:
Is this the only place in the codebase where this logic is done?  It seems like it would be prone to possibly change in the future (if we end up needing another way to validate requests).  If the logic to determine what a "valid un-accepted request" is is duplicated, we might want to abstract that into a single method on a service somewhere.

Reply:
IMO it can be moved directly into the login request model and have a `IsRequestValid` property or method

Reply:
Agree with @fedemkr the model can handle it on its own.

Reply:
But this is valid/invalid for the current context because we want unanswered requests. I'll create two properties, one for expired and another for answered.